tensorflow_addons/losses/contrastive_test.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for contrastive loss."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from tensorflow_addons.losses import contrastive
+from tensorflow_addons.utils import test_utils
+
+
+# pylint:disable=not-callable
+@test_utils.run_all_in_graph_and_eager_modes
+class ContrastiveLossTest(tf.test.TestCase):
+    def test_config(self):
+        cl_obj = contrastive.ContrastiveLoss(
+            reduction=tf.keras.losses.Reduction.SUM, name='cl')
+        self.assertEqual(cl_obj.name, 'cl')
+        self.assertEqual(cl_obj.reduction, tf.keras.losses.Reduction.SUM)
+
+    def test_all_correct_unweighted(self):
```

Comment:
If `y_pred = 1`, the loss would try to minimize the distance to 0 (closer); otherwise, the loss try to maximize the distance to `margin`. That's why I call it `all_correct`, meaning that we can obtain minimum loss in this test case. (a quite confusing though).

Reply:
Good point! It might be better to keep consistent with [keras: losses_test.py](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/losses_test.py): all_correct means that  y_true = y_pred (distance). Or choose a new name?

Reply:
Seems that test cases in keras called `*all_correct*` coincide with zero loss. I'd vote for a new name, say `test_zero_loss`, for distinguishing.

Reply:
Good name!